src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php

```diff
@@ -184,11 +178,9 @@ protected function echoLine($line, $depth, $indentPad)
     /**
      * Converts a non-UTF-8 string to UTF-8.
      *
-     * @param string $s The non-UTF-8 string to convert
-     *
-     * @return string The string converted to UTF-8
+     * @return string|null The string converted to UTF-8
      */
-    protected function utf8Encode($s)
+    protected function utf8Encode(?string $s)
     {
         if (preg_match('//u', $s)) {
```

Comment:
if $s is nullable, we call preg_match with null which would fail in strict mode. in this case we can skip it.

Reply:
I could change this to

```php
if (null === $s || preg_match('//u', $s)) {
```

And I would do this on 3.4 already and fix the doc block there. WDYT?

Reply:
Oh, I just noticed that @nicolas-grekas did exactly that with #33179. 🙈